### ISSUE TITLE
2 tests fail when run build.cmd

### ISSUE THREAD

**Issue by cnblogs-dudu** [dotnet/corefx#560]
Discovering: MemoryMappedFile.Tests
Discovered:  MemoryMappedFile.Tests
Starting:    MemoryMappedFile.Tests
Err445! Got unexpected exception: System.IO.IOException: There is not enough space on the disk.
  
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 45
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 107
     at CreateViewAccessor.RunTestLargeCapacity(Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewAccessor.cs:line 378
  Err445! Got unexpected exception: System.IO.IOException: There is not enough space on the disk.
  
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 45
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 107
     at CreateViewAccessor.RunTestLargeCapacity(Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewAccessor.cs:line 378
  Err445! Got unexpected exception: System.IO.IOException: There is not enough space on the disk.
  
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 45
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 107
     at CreateViewStream.RunTestLargeCapacity(Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewStream.cs:line 377
  Err445! Got unexpected exception: System.IO.IOException: There is not enough space on the disk.
  
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 45
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 107
     at CreateViewAccessor.RunTestLargeCapacity(Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewAccessor.cs:line 378
  FAiL! iCountErrors==3
  Err445! Got unexpected exception: System.IO.IOException: There is not enough space on the disk.
  
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore(SafeFileHandle fileHandle, String mapName, HandleInheritability inheritability, MemoryMappedFileAccess access, MemoryMappedFileOptions options, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 45
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180
     at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 107
     at CreateViewStream.RunTestLargeCapacity(Int64 capacity) in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewStream.cs:line 377
     CreateViewAccessor.CreateViewAccessorTestCases [FAIL]
        One or more test cases failed.
        Stack Trace:
           e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewAccessor.cs(32,0): at CreateViewAccessor.CreateViewAccessorTestCases()
  Finished:    System.Diagnostics.FileVersionInfo.Tests

=== TEST EXECUTION SUMMARY ===
     System.Diagnostics.FileVersionInfo.Tests  Total: 6, Errors: 0, Failed: 0, Skipped: 0, Time: 2.371s
  FAiL! iCountErrors==2
     CreateViewStream.CreateViewStreamTestCases [FAIL]
        One or more test cases failed.
        Stack Trace:
           e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewStream.cs(31,0): at CreateViewStream.CreateViewStreamTestCases()
  Finished:    MemoryMappedFile.Tests

=== TEST EXECUTION SUMMARY ===
     MemoryMappedFile.Tests  Total: 7, Errors: 0, Failed: 2, Skipped: 0, Time: 1.345s

**Comment by krwq** [comment]
@cnblogs-dudu could give some more details?
What is your OS?
How much space do you have left on your E: drive?
Does it happens all the time or intermittently?
How did you run the tests? (did you use build.cmd or VS)

@KrzysztofCwalina, @mellinoe is it known issue?

I will send a PR to disable those tests in few minutes.

**Comment by mellinoe** [comment]
The tests try to make files with the following sizes, so I'd imagine they probably fail if your disk is too full:
````CSharp
Int64[] capacities = { 2147483647, 2147483648, 2147483649, 4294967295, 4294967296, 4294967297 };
````
We can probably either tone down these numbers or just make such a case deliver a warning, maybe?

**Comment by cnblogs-dudu** [comment]
@krwq 
1) Windows Server 2012
2) Free Space: 4.81G
3) all the time
4) usr build.com

**Comment by cnblogs-dudu** [comment]
@mellinoe 
failed when capacity is 4294967296

**Comment by krwq** [comment]
@cnblogs-dudu, does it fail for 4294967295 too? Are you using 64bit OS or 32? What type of file system is your E:? Is it FAT32?

**Comment by cnblogs-dudu** [comment]
@krwq 
fail too. 
64bit. 
NTFS. 

```
4294967295
System.IO.IOException: There is not enough space on the disk.

at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateCore
(SafeFileHandle fileHandle,   String mapName,   HandleInheritability inheritability,   MemoryMappedFileAccess access,   MemoryMappedFileOptions options,  Int64 capacity) 
in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.Windows.cs:line 45

at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) 
in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 180

at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity) 
in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\src\System\IO\MemoryMappedFiles\MemoryMappedFile.cs:line 107

at CreateViewAccessor.RunTestLargeCapacity(Int64 capacity) 
in e:\Dev\GitHub\DotNet\corefx\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile\CreateViewAccessor.cs:line 379
```

**Comment by cnblogs-dudu** [comment]
@krwq 
When increased free space from 4.81G to 9.21G, "There is not enough space on the disk" exception was gone.

**Comment by krwq** [comment]
Ok, we should mark all tests testing large files as outer loop as creating 4GB files feels like an overkill for the unit test. In this case it looks like we might need to refactor the code to do so first.

**Comment by stephentoub** [comment]
Fixed by #2334
